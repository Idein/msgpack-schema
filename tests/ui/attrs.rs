use msgpack_schema::*;

#[derive(Serialize)]
#[tag = 1]
struct S1 {
    x: String,
}

#[derive(Serialize)]
#[optional]
struct S2 {
    x: String,
}

#[derive(Serialize)]
#[untagged]
struct S3(String);

#[derive(Serialize)]
#[untagged]
struct S4;

#[derive(Deserialize)]
#[tag = 1]
struct S5 {
    x: String,
}

#[derive(Deserialize)]
#[optional]
struct S6 {
    x: String,
}

#[derive(Deserialize)]
#[untagged]
struct S7(String);

#[derive(Deserialize)]
#[untagged]
struct S8;

fn main() {}
