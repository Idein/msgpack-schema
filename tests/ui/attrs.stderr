error: #[tag] at an invalid position
 --> $DIR/attrs.rs:4:1
  |
4 | #[tag = 1]
  | ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/attrs.rs:10:1
   |
10 | #[optional]
   | ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/attrs.rs:16:1
   |
16 | #[untagged]
   | ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/attrs.rs:20:1
   |
20 | #[untagged]
   | ^^^^^^^^^^^

error: #[tag] at an invalid position
  --> $DIR/attrs.rs:24:1
   |
24 | #[tag = 1]
   | ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/attrs.rs:30:1
   |
30 | #[optional]
   | ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/attrs.rs:36:1
   |
36 | #[untagged]
   | ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/attrs.rs:40:1
   |
40 | #[untagged]
   | ^^^^^^^^^^^
